### ISSUE TITLE
Timeshift not working with latest nightly kodi

### ISSUE THREAD

**Issue by suncore** [kodi-pvr/pvr.hts#372]
Pausing live TV does not work any more with the latest nightly Kodi. 
A thread in the forum suggests this addon needs to be changed. See https://forum.kodi.tv/showthread.php?tid=334971&pid=2762416
It says "The add-on must implement the PVR add-on API function GetStreamTimes." (That was for mythtv but it seems plausible it holds for this addon also).

**Comment by ksooo** [comment]
This addon already implements this function.

**Comment by suncore** [comment]
OK, Thanks. Probably I just need to update the plugin or continue troubleshooting.

**Comment by ksooo** [comment]
The addon implements this function for many months. So I doubt that updating it would change anything.

Do you have actually the problem that timeshift does not work with recent kodi+pvr.hts+tvheadend or is it just theory, because otehrs reported problems.

I'm investigating another case where a user reported that timeshifting does not work any longer and I have an idea what could go wrong.

**Comment by ksooo** [comment]
Please post a screenshot of the timeshift settings of the involved tvheadend server.

**Comment by suncore** [comment]
Pausing live TV has stopped working for me at home. It just plains ignore the pause button.

**Comment by suncore** [comment]
Attaching screenshot 
![timeshift settings](https://user-images.githubusercontent.com/781533/44709599-501c6300-aaaa-11e8-83b6-91801f371d2d.png)

**Comment by suncore** [comment]
Yes that helped. Unticking on-demand enables working pause/timeshift. 
I also noticed that if on-demand is enabled, the TV OSD does not even show a pause button, only stop and record, so the HTS PVR client addon does not even "know" that pausing is available.

**Comment by ksooo** [comment]
Thanks, I now know what the problem is. Give me some time to think about a solution. Not sure whether this best to fix in the addon or kodi core, atm.

I will reopen the issue and adapt the title to track the "on-demand timeshift" problem

**Comment by perlbo** [comment]
Thank you very much.
